core/chain_manager.go

```diff
@@ -419,7 +422,25 @@ func (self *ChainManager) GetBlock(hash common.Hash) *types.Block {
 		glog.V(logger.Error).Infof("invalid block RLP for hash %x: %v", hash, err)
 		return nil
 	}
-	return (*types.Block)(&block)
+
+	blk := (*types.Block)(&block)
```

Comment:
This needs to be moved to its own separate function. We want to limit the amount of database gets and RLP decoding required to get a block. Most cases when `GetBlock` is called you don't require the receipts.

Move this to an additional method like

```go
func (cm *ChainManager) GetReceipts(block *types.Block) types.Receipts {
    // your code
}
```